tikv-client/src/main/java/com/pingcap/tikv/meta/TiDAGRequest.java

```diff
@@ -952,76 +955,114 @@ private void init() {
     init(hasIndex());
   }
 
-  public IndexScanType getIndexScanType() {
+  public ScanType getScanType() {
     if (hasIndex()) {
       if (isDoubleRead) {
-        return IndexScanType.INDEX_SCAN;
+        return ScanType.INDEX_LOOKUP;
       } else {
-        return IndexScanType.COVERING_INDEX_SCAN;
+        return ScanType.INDEX_READER;
       }
     } else {
-      return IndexScanType.TABLE_SCAN;
+      return ScanType.TABLE_READER;
     }
   }
 
   @Override
   public String toString() {
-    return this.copy().toStringInternal();
+    return this.toStringInternal();
   }
 
   private String toStringInternal() {
-    init();
```

Comment:
why move the init method to another place.
If init isn't invoked at the beginning, then it's maybe a inappropriate name

Reply:
This method is used to initialize some variables. There is no place to use it anymore, maybe this method can be removed.